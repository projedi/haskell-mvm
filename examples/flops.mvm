print(1.0 + 2.0, 1 + 0.99, 1.2 + 3, 0.01 + -0.99);
print(1.4 * 3.5, 6 * 0.5, 0.23 * 7, 1.2 * -0.45);
print(6 / 2, 1 / 3, 1.5 / 0.5);

print(0.0 == 0.0, 0.0 == -0.0, 0.0 == 1.0, 1.0 == 0.0);
print(0.0 != 0.0, 0.0 != -0.0, 0.0 != 1.0, 1.0 != 0.0);
print(0.0 < 0.0, 0.0 < -0.0, 0.0 < 1.0, 1.0 < 0.0);
print(0.0 <= 0.0, 0.0 <= -0.0, 0.0 <= 1.0, 1.0 <= 0.0);
print(0.0 > 0.0, 0.0 > -0.0, 0.0 > 1.0, 1.0 > 0.0);
print(0.0 >= 0.0, 0.0 >= -0.0, 0.0 >= 1.0, 1.0 >= 0.0);
