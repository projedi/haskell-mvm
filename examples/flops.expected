3 1.99 4.2 -0.98
4.9 3 1.61 -0.54
3 0 3 0.333333 0.333333
1 1 0 0
0 0 1 1
0 0 1 0
1 1 1 0
0 0 0 1
1 1 0 1
